titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/endpoint/v3/grpc/DefaultJobManagementServiceGrpc.java

```diff
@@ -345,38 +348,31 @@ public void findTask(TaskId request, StreamObserver<Task> responseObserver) {
     @Override
     public void updateJobCapacity(JobCapacityUpdate request, StreamObserver<Empty> responseObserver) {
         execute(callMetadataResolver, responseObserver, callMetadata -> {
-            com.netflix.titus.api.jobmanager.model.job.Job<ServiceJobExt> job = jobOperations
-                    .getJob(request.getJobId())
-                    .map(j -> {
-                        if (!JobFunctions.isServiceJob(j)) {
-                            throw JobManagerException.notServiceJob(j.getId());
-                        }
-                        return (com.netflix.titus.api.jobmanager.model.job.Job<ServiceJobExt>) j;
-                    })
-                    .orElseThrow(() -> JobManagerException.jobNotFound(request.getJobId()));
-
+            final com.netflix.titus.api.jobmanager.model.job.Job<ServiceJobExt> serviceJobExtJob = verifyAndGetServiceJob(request.getJobId());
             com.netflix.titus.api.jobmanager.model.job.Capacity newCapacity = V3GrpcModelConverters.toCoreCapacity(request.getCapacity());
+            sanitizeAndUpdateJobCapacity(request.getJobId(), newCapacity, serviceJobExtJob.getJobDescriptor(), callMetadata, responseObserver);
+        });
+    }
 
-            Set<ValidationError> violations = CollectionsExt.merge(
-                    entitySanitizer.validate(newCapacity),
-                    validateCustomJobLimits(JobFunctions.changeServiceJobCapacity(job.getJobDescriptor(), newCapacity))
-            );
-            if (!violations.isEmpty()) {
-                safeOnError(logger, TitusServiceException.invalidArgument(violations), responseObserver);
-                return;
-            }
+    @Override
+    public void updateJobCapacityWithOptionalAttributes(JobCapacityUpdateWithOptionalAttributes request, StreamObserver<Empty> responseObserver) {
+        execute(callMetadataResolver, responseObserver, callMetadata -> {
+            final com.netflix.titus.api.jobmanager.model.job.Job<ServiceJobExt> serviceJobExtJob = verifyAndGetServiceJob(request.getJobId());
+            final Capacity currentCapacity = serviceJobExtJob.getJobDescriptor().getExtensions().getCapacity();
```

Comment:
you can just do `getCapacity().toBuilder()` to shorten these lines on modifying the current capacity